Library/Homebrew/cmd/update-bash.sh

```diff
@@ -0,0 +1,256 @@
+}
+
+reset_on_interrupt() {
+  [ -z "$INITIAL_BRANCH" ] || git checkout "$INITIAL_BRANCH"
+  git reset --hard "$INITIAL_REVISION" $QUIET_ARGS
+  if [ -n "$INITIAL_BRANCH" ]
+  then
+    pop_stash
+  else
+    pop_stash_message
+  fi
+}
+
+pull() {
+  cd "$DIR" || return
+  TAP_VAR=$(repo_var "$DIR")
+  unset STASHED
+
+  # The upstream repository's default branch may not be master;
+  # check refs/remotes/origin/HEAD to see what the default
+  # origin branch name is, and use that. If not set, fall back to "master".
+  INITIAL_BRANCH="$(git symbolic-ref --short HEAD 2>/dev/null)"
+  UPSTREAM_BRANCH="$(upstream_branch)"
+  export HOMEBREW_UPDATE_AFTER"$TAP_VAR"="$(git rev-parse "$UPSTREAM_BRANCH")"
+
+  if [ -n "$(git status --untracked-files=all --porcelain 2>/dev/null)" ]
+  then
+    if [ -n "$HOMEBREW_VERBOSE" ]
+    then
+      echo "Stashing uncommitted changes to $DIR."
+      git status --short --untracked-files=all
+    fi
+    git -c "user.email=brew-update@localhost" \
+        -c "user.name=brew update" \
+        stash save --include-untracked $QUIET_ARGS
+    git reset --hard $QUIET_ARGS
+    STASHED="1"
+  fi
+
+  # Used for testing purposes, e.g., for testing formula migration after
+  # renaming it in the currently checked-out branch. To test run
+  # "brew update --simulate-from-current-branch"
+  if [ -n "$HOMEBREW_SIMULATE_FROM_CURRENT_BRANCH" ]
+  then
+    INITIAL_REVISION="$(git rev-parse -q --verify "$(upstream_branch)")"
+    CURRENT_REVISION="$(read_current_revision)"
+    if ! git merge-base --is-ancestor "$INITIAL_REVISION" "$CURRENT_REVISION"
+    then
+      echo "Your HEAD is not a descendant of $UPSTREAM_BRANCH!" >&2
+      exit 1
+    fi
+    return
+  fi
+
+  if [ "$INITIAL_BRANCH" != "$UPSTREAM_BRANCH" ] && [ -n "$INITIAL_BRANCH" ]
+  then
+    git checkout --force -B "$UPSTREAM_BRANCH" "origin/$UPSTREAM_BRANCH" $QUIET_ARGS
+  fi
+
+  INITIAL_REVISION="$(read_current_revision)"
+
+  # ensure we don't munge line endings on checkout
+  git config core.autocrlf false
+
+  trap reset_on_interrupt SIGINT
+
+  if [ -n "$HOMEBREW_REBASE" ]
+  then
+    git rebase $QUIET_ARGS "origin/$UPSTREAM_BRANCH"
+  else
+    git merge --no-edit --ff $QUIET_ARGS "origin/$UPSTREAM_BRANCH"
+  fi
+
+  trap - SIGINT
+
+  CURRENT_REVISION="$(read_current_revision)"
+
+  if [ "$INITIAL_BRANCH" != "$UPSTREAM_BRANCH" ] && [ -n "$INITIAL_BRANCH" ]
+  then
+    git checkout "$INITIAL_BRANCH" $QUIET_ARGS
+    pop_stash
+  else
+    pop_stash_message
+  fi
+}
+
+update() {
+  if [ -z "$HOMEBREW_DEVELOPER" ]
+  then
+    echo "This command is currently only for Homebrew developers' use." >&2
+    exit 1
+  fi
+
+  for i in "$@"
+  do
+    case "$i" in
+      update|update-bash) shift ;;
+      --help) exec brew update --help ;;
+      --verbose) HOMEBREW_VERBOSE=1 ;;
+      --rebase) HOMEBREW_REBASE=1 ;;
+      --simulate-from-current-branch) HOMEBREW_SIMULATE_FROM_CURRENT_BRANCH=1 ;;
+      --*) ;;
+      -*v*) HOMEBREW_VERBOSE=1 ;;
+      *)
```

Comment:
There's a case missing directly before this one, namely to handle other single-letter options: `-*) ;;`. Currently, `brew update-bash -d` results in:

```
This command updates brew itself, and does not take formula names.
Use 'brew upgrade <formula>'.
```